crates/typst/src/math/row.rs

```diff
@@ -171,35 +176,33 @@ impl MathRow {
             TIGHT_LEADING.at(font_size)
         };
 
-        let mut rows: Vec<_> = self.rows();
-
-        if matches!(rows.last(), Some(row) if row.0.is_empty()) {
-            rows.pop();
-        }
-
-        let AlignmentResult { points, width } = alignments(&rows);
-        let mut frame = Frame::soft(Size::zero());
-
+        let mut frames: Vec<(Frame, Point)> = vec![];
+        let mut size = Size::zero();
         for (i, row) in rows.into_iter().enumerate() {
-            let sub = row.into_line_frame(&points, align);
-            let size = frame.size_mut();
+            if i == row_count - 1 && row.0.is_empty() {
+                continue;
+            }
+
+            let sub = row.into_line_frame(&alignments.points, align);
+            let sub_size = sub.size();
             if i > 0 {
                 size.y += leading;
             }
 
             let mut pos = Point::with_y(size.y);
-            if points.is_empty() {
-                pos.x = align.position(width - sub.width());
```

Comment:
Is there a particular reason to change `sub.width()` to `sub_size.x`? I find the former easier to read.